pkg/codegen/nodejs/gen.go

```diff
@@ -1136,12 +1147,24 @@ func (mod *modContext) genIndex(exports []string) string {
 		}
 	}
 
-	var children []string
+	children := codegen.NewStringSet()
+
 	for _, mod := range mod.children {
-		children = append(children, mod.mod)
+		child := mod.mod
+		if mod.compatibility == kubernetes20 {
+			// Trim version suffix from child modules. Nested versions will have their own index.ts file.
+			if match, _ := regexp.MatchString(`/v\d+((alpha|beta)\d+)?`, child); match {
+				if i := strings.LastIndex(child, "/"); i > 0 {
+					child = child[:i]
+				}
+			}
```

Comment:
I'd suggest compiling the regex and storing it as a package variable (also, if it's possible to anchor it to the end of the string, I'd add that):

```go
var k8sVersionRegex = regexp.MustCompile(`/v\d+((alpha|beta)\d+)?$`)
```

Then you can use `FindStringIndex` and avoid `strings.LastIndex`:

```suggestion
			if match := k8sVersionRegex.FindStringIndex(child); len(match) != 0 {
				child = child[:match[0]]
			}
```